.github/workflows/gofmt.yml

```diff
@@ -0,0 +1,33 @@
+name: gofmt
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    paths:
+      - "**/*.go"
+
+jobs:
+  run_gofmt:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+        with:
+          cache: false
+          go-version: '^1.21'
+
+      - name: gofmt
+        run: |
+          GOFMT_OUTPUT="$(gofmt -l .)"
+          # Currently no simple way to get a non-0 exit code directly from
+          # gofmt
+          if [ -n "$GOFMT_OUTPUT" ]; then 
+            echo "The following files are not formatted properly:" >&2
+            echo "$GOFMT_OUTPUT" >&2
```

Comment:
Happy to do this a different way if you have a better way to do this.

Some loose inspiration from https://github.com/Jerome1337/gofmt-action (MIT licensed), but figured this is simple enough to do that it's overkill to include a third party action, and I've made some adjustments anyway.